core/src/main/java/org/infinispan/LockedStream.java

```diff
@@ -135,6 +136,42 @@ default void forEach(SerializableBiConsumer<Cache<K, V>, ? super CacheEntry<K, V
       return invokeAll((BiFunction<Cache<K, V>, ? super CacheEntry<K, V>, R>) biFunction);
    }
 
+   /**
+    * {@inheritDoc}
+    */
+   @Override
+   LockedStream sequentialDistribution();
```

Comment:
All return types need to be in the generic form LockedStream<K, V> to preserve type information in chained invocations

Reply:
Oops I didn't mean to forget those 👍